job-service-db/src/main/resources/db/migration/V3__job_service.sql

```diff
@@ -0,0 +1,233 @@
+
+ALTER TABLE job_task_data
+    DROP CONSTRAINT IF EXISTS fk_job_task_data CASCADE;
+
+ALTER TABLE job_task_data
+    DROP CONSTRAINT IF EXISTS pk_job_task_data CASCADE;
+
+ALTER TABLE job_task_data
+    ADD CONSTRAINT fk_job_task_data FOREIGN KEY (partition_id, job_id) REFERENCES job (partition_id, job_id);
+
+ALTER TABLE job_task_data
+    ADD CONSTRAINT pk_job_task_data PRIMARY KEY (partition_id, job_id);
+
+-------- public."label"
+
+CREATE TABLE IF NOT EXISTS public."label"
+(
+    partition_id varchar(40)  NOT NULL DEFAULT 'default'::character varying,
+    job_id       varchar(48)  NOT NULL,
+    "label"      varchar(255) NOT NULL,
+    value        varchar(255) NULL,
+    CONSTRAINT label_pkey PRIMARY KEY (partition_id, job_id, label),
+    CONSTRAINT fk_label_job FOREIGN KEY (partition_id, job_id) REFERENCES public.job (partition_id, job_id)
+);
+
+-------- public.stowed_task
+
+CREATE TABLE IF NOT EXISTS public.stowed_task
+(
+    partition_id                               varchar(40)  NOT NULL,
+    job_id                                     varchar(48)  NOT NULL,
+    task_classifier                            varchar(255) NOT NULL,
+    task_api_version                           int4         NOT NULL,
+    task_data                                  bytea        NOT NULL,
+    task_status                                varchar(255) NOT NULL,
+    context                                    bytea        NOT NULL,
+    "to"                                       varchar(255) NOT NULL,
+    tracking_info_job_task_id                  varchar(255) NOT NULL,
+    tracking_info_last_status_check_time       int8         NULL,
+    tracking_info_status_check_interval_millis int8         NULL,
+    tracking_info_status_check_url             text         NULL,
+    tracking_info_tracking_pipe                varchar(255) NULL,
+    tracking_info_track_to                     varchar(255) NULL,
+    source_info                                bytea        NULL,
+    correlation_id                             varchar(255) NULL,
+    CONSTRAINT fk_stowed_task FOREIGN KEY (partition_id, job_id) REFERENCES public.job (partition_id, job_id)
+);
+
+-- Drop Indexes
+DROP INDEX IF EXISTS idx_job_jobid_status;
+
+-- Create Indexes
+CREATE INDEX IF NOT EXISTS idx_deleted_parent_table_log
+    ON public.deleted_parent_table_log
+        USING btree (table_name);
+
+CREATE INDEX IF NOT EXISTS idx_delete_log_table_name
+    ON public.delete_log
+        USING btree (table_name);
+
+CREATE INDEX IF NOT EXISTS idx_job_create_date
+    ON public.job
+        USING btree (create_date);
+
+CREATE INDEX IF NOT EXISTS idx_partition_id_and_job_id
+    ON public.stowed_task
+        USING btree (partition_id, job_id);
+
+CREATE INDEX IF NOT EXISTS idx_job_partition_id_and_dependent_job_id
+    ON public.job_dependency
+        USING btree (partition_id, dependent_job_id);
+
+CREATE OR REPLACE FUNCTION internal_to_regclass(rel_name VARCHAR(63))
```

Comment:
I just made a comment about adding forward declarations to the modules where functions are used but I guess an alternative would be to forward declare in the versioned migrations.  If we are going to do that then we'd need to declare all internal functions here and not just this one, and also don't put the real implementations in, since we don't want it in two places.